File: frontend/src/utils/dateUtils.ts
import dayjs, { Dayjs } from 'dayjs';

export function isSaturday(date: Dayjs): boolean {
  return date.day() === 6;
}

export function nextSaturday(from: Dayjs = dayjs()): Dayjs {
  const day = from.day();
  const daysUntilSat = day <= 6 ? (6 - day) || 7 : 6;
  return from.add(daysUntilSat, 'day');
}

export function disableNonSaturday(current: Dayjs): boolean {
  return current.day() !== 6;
}

export function formatDate(date: string): string {
  return dayjs(date).format('YYYY-MM-DD');
}

export function formatDateCN(date: string): string {
  const d = dayjs(date);
  return `${d.month() + 1}月${d.date()}日`;
}

const REGION_MAP: Record<string, string> = {
  west: '美西',
  central: '美中',
  east: '美东',
};

export function regionLabel(region: string): string {
  return REGION_MAP[region] || region;
}

File: backend/app/utils/date_utils.py
from datetime import date, timedelta


def validate_saturday(ship_date: date) -> bool:
    """Check if a date is Saturday (weekday() == 5)."""
    return ship_date.weekday() == 5


def next_saturday(from_date: date) -> date:
    """Get the next Saturday from a given date."""
    days_ahead = 5 - from_date.weekday()
    if days_ahead <= 0:
        days_ahead += 7
    return from_date + timedelta(days=days_ahead)


def format_unit_label(ship_date: date, region: str) -> str:
    """Format shipment unit label: '{月}月{日}日-{仓库}'."""
    region_map = {"west": "美西", "central": "美中", "east": "美东"}
    region_label = region_map.get(region, region)
    return f"{ship_date.month}月{ship_date.day}日-{region_label}"
